### DISCUSSION TITLE
`CallerMemberName` and local methods

### DISCUSSION THREAD

**Discussion by FaustVX** [dotnet/csharplang#6486] [General] [answered]
Why the compiler inserts the parent method identifier instead of the local method indentifier here ?
```C#
static class Program
{
    static void Main()
    {
        Print(); // will print Main
        Level(); // will also print Main
        
        static void Level()
            => Print();

        static void Print([CallerMemberName]string name = null!)
            => Console.WriteLine(name);
    }
}
```
[sharplab.io](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQGUawJYGMCCwyMCUuMAshALbAQLICSAdizPWGSgDQwgJxm3ACYgA1AB8AAgCYAjAFgAUFIAMAAilyAdACVBMbNQjaAwgHtqAB2wAbeunoA3PCgDcy5VoBsmmZrkAdmUAb2V1CIDfKQAWdQoobGYACgBKcMiwpUic9QAFBCSYNI9s3IiAGQgnCFsSjPKG3J9NOKqauvSy8qzy8oKi+u7cgF9PYZyW2PzC5mKAbVMoW3sEKlp6ADkoYwBdLQ1mHYh1AF51ZjgVgEIuvtzTgD4AgE5ko+NU0vumsaU/5RAA)

Of course, it works the same with top-level program, but will print `<Main>$` instead.

**Comment by RikkiGibson** [comment, marked as answer]
I think the idea here is that local functions are not *members* and thus their names don't get used for CallerMemberName. Only the containing member's name gets used, i.e., Main.

Since the containing member will always be the same for all calls of a local function, the current behavior is perhaps not that useful.

**Reply by FaustVX** [reply]
Ok,
Is it possible to change that in a future version ? (or is this a breaking change ?)
To me, it's not the expected behavior, and like you said, it's not useful.

**Reply by CyrusNajmabadi** [reply]
Definitely a breaking change :-). I technically prefer the current behavior. Otherwise it's visible if your caller is a lambda vs local function, which I don't like.

**Reply by FaustVX** [reply]
Ah, I didn't think of that 😄
